### PR TITLE
Add support for ZKR accelerators

### PR DESCRIPTION
**Pull request by shkoo**
* Add new execute_zkr system call to invoke a zkr
* Add zkr registry to allow guests to specify zkr by control id

### CODE REVIEW

**Comment by flaub** [comment]
This is great. Could you add a test that uses this new functionality?

**Comment by shkoo** [comment]
> This is great. Could you add a test that uses this new functionality?

Test is in risc0/risczero-wip#612

**Comment by flaub** [comment]
OK, could you add some tests for `risc0`? I think especially for the unconstrained wrapper and for the new syscalls.

File: risc0/zkvm/methods/guest/src/bin/multi_test.rs [hunk 1]
```diff
@@ -379,5 +379,45 @@ fn main() {
...
+            env::log("Starting running unconstrained");
+
+            let f = || {
+                // Unfortunately we can't use env::cycle_count since
+                // it's always zero when running unconstrained.
+
+                const CYCLES_PER_LOOP: u32 = 2; // Determined empirically
+
+                for _ in 0..cycles_left / CYCLES_PER_LOOP {
+                    unsafe { asm!("") }
```
Comment (flaub) [inline risc0/zkvm/methods/guest/src/bin/multi_test.rs:397]: What does this do? could you add a comment?

**Review by flaub — APPROVED** [review]

**Pull request merged**